src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +832,56 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool   $boolean
+     * @param string $template Template message to use if the definition is deprecated
+     *
+     * @return Definition the current instance
+     *
+     * @todo Support a suggested definition instead ?
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true, $template = 'The service %s relies on a deprecated definition. You should stop using it.')
```

Comment:
I think I got confused with the fact that a service basically represents a Definition (well, at least in the given loaders), as a service is just an id basically, pointing to a definition. After all, it is the definition that supports the deprecated status ?

I guess I choose my words poorly for the message. And the worst thing is, I had a `The service %s is deprecated` before... heh.